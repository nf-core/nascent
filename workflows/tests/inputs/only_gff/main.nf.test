nextflow_pipeline {

    name "GFF"
    script "../../../../main.nf"
    tag "input"
    tag "gff"

    test("Should run with only a GFF file") {

        when {
            params {
                outdir = "$outputDir"
                gff = 'https://raw.githubusercontent.com/nf-core/test-datasets/nascent/reference/genes_chr21.gff'
                gtf = null
                bed = null
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml")).match("software_versions") },
                { assert snapshot(
                    workflow.trace.tasks().size(),
                    UTILS.getAllFilesFromDir("$outputDir/preprocessing/fastp/", ".json"),
                    bam("$outputDir/bwa/cd4_REP1.sorted.bam").getSamLinesMD5(),
                    bam("$outputDir/bwa/cd4_REP2.sorted.bam").getSamLinesMD5(),
                    bam("$outputDir/bwa/jurkat.sorted.bam").getSamLinesMD5(),
                    path("$outputDir/bwa/samtools_stats").list(),
                    path("$outputDir/quality_control/bbsplit").list(),
                    UTILS.getAllFilesFromDir("$outputDir/quality_control/preseq", ".txt"),
                    path("$outputDir/quality_control/rseqc/read_duplication/xls").list(),
                    path("$outputDir/quality_control/rseqc/read_duplication/rscript").list(),
                    path("$outputDir/quality_control/rseqc/infer_experiment").list(),
                    path("$outputDir/quality_control/rseqc/read_distribution").list(),
                    path("$outputDir/coverage_graphs/").list(),
                    path("$outputDir/transcript_identification/homer/cd4.bed"),
                    path("$outputDir/transcript_identification/homer/jurkat.bed"),
                    // FIXME Not determinstic because of the order of files
                    // Add to the other tests when fixed
                    // UTILS.getAllFilesFromDir("$outputDir/transcript_identification/pints/", ".bed"),
                    path("$outputDir/transcript_identification/intersect/").list(),
                    path("$outputDir/transcript_identification/filtered/").list(),
                    // FIXME Not determinstic because of the order of files
                    // Add to the other tests when fixed
                    // path("$outputDir/quantification/").list(),
                    path("$outputDir/multiqc/multiqc_report.html").exists(),
                ).match("output_files")
                }
            )
        }
    }
}
