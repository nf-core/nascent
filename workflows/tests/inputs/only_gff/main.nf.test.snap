{
    "Should run with only a GFF file": {
        "content": [
            93,
            {
                "BBMAP_PILEUP": {
                    "bbmap": 39.01,
                    "samtools": "1.16.1",
                    "pigz": 2.6
                },
                "BEDTOOLS_GENOMECOV_MINUS": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_GENOMECOV_PLUS": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_INTERSECT": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_INTERSECT_FILTER": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_MERGE": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_SORT": {
                    "bedtools": "2.31.1"
                },
                "BWA_INDEX": {
                    "bwa": "0.7.18-r1243-dirty"
                },
                "BWA_MEM": {
                    "bwa": "0.7.18-r1243-dirty",
                    "samtools": 1.2
                },
                "CAT_CAT": {
                    "pigz": "2.3.4"
                },
                "CUSTOM_GETCHROMSIZES": {
                    "getchromsizes": 1.2
                },
                "DEEPTOOLS_BAMCOVERAGE_MINUS": {
                    "deeptools": "3.5.5"
                },
                "DEEPTOOLS_BAMCOVERAGE_PLUS": {
                    "deeptools": "3.5.5"
                },
                "FASTP": {
                    "fastp": "0.23.4"
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GFFREAD": {
                    "gffread": "0.12.7"
                },
                "GTF2BED": {
                    "perl": "5.26.2"
                },
                "HOMER_MAKETAGDIRECTORY": {
                    "homer": 4.11,
                    "samtools": 1.21
                },
                "PINTS_CALLER": {
                    "python": "3.12.6",
                    "pints": "1.1.13"
                },
                "PRESEQ_CCURVE": {
                    "preseq": "3.1.1"
                },
                "PRESEQ_LCEXTRAP": {
                    "preseq": "3.1.1"
                },
                "RSEQC_INFEREXPERIMENT": {
                    "rseqc": "5.0.2"
                },
                "RSEQC_READDISTRIBUTION": {
                    "rseqc": "5.0.2"
                },
                "RSEQC_READDUPLICATION": {
                    "rseqc": "5.0.2"
                },
                "SAMTOOLS_MERGE": {
                    "samtools": 1.21
                },
                "SUBREAD_FEATURECOUNTS_GENE": {
                    "subread": "2.0.1"
                },
                "SUBREAD_FEATURECOUNTS_PREDICTED": {
                    "subread": "2.0.1"
                },
                "Workflow": {
                    "nf-core/nascent": "v2.3.0"
                }
            },
            [
                "bwa",
                "bwa/cd4_REP1.sorted.bam",
                "bwa/cd4_REP1.sorted.bam.bai",
                "bwa/cd4_REP2.sorted.bam",
                "bwa/cd4_REP2.sorted.bam.bai",
                "bwa/jurkat.sorted.bam",
                "bwa/jurkat.sorted.bam.bai",
                "bwa/samtools_stats",
                "bwa/samtools_stats/cd4_REP1.sorted.bam.flagstat",
                "bwa/samtools_stats/cd4_REP1.sorted.bam.idxstats",
                "bwa/samtools_stats/cd4_REP1.sorted.bam.stats",
                "bwa/samtools_stats/cd4_REP2.sorted.bam.flagstat",
                "bwa/samtools_stats/cd4_REP2.sorted.bam.idxstats",
                "bwa/samtools_stats/cd4_REP2.sorted.bam.stats",
                "bwa/samtools_stats/jurkat.sorted.bam.flagstat",
                "bwa/samtools_stats/jurkat.sorted.bam.idxstats",
                "bwa/samtools_stats/jurkat.sorted.bam.stats",
                "cat",
                "cat/cd4.bed",
                "cat/jurkat.bed",
                "coverage_graphs",
                "coverage_graphs/cd4_REP1.dreg.bedGraph",
                "coverage_graphs/cd4_REP1.minus.bedGraph",
                "coverage_graphs/cd4_REP1.minus.bigWig",
                "coverage_graphs/cd4_REP1.plus.bedGraph",
                "coverage_graphs/cd4_REP1.plus.bigWig",
                "coverage_graphs/cd4_REP2.dreg.bedGraph",
                "coverage_graphs/cd4_REP2.minus.bedGraph",
                "coverage_graphs/cd4_REP2.minus.bigWig",
                "coverage_graphs/cd4_REP2.plus.bedGraph",
                "coverage_graphs/cd4_REP2.plus.bigWig",
                "coverage_graphs/jurkat.dreg.bedGraph",
                "coverage_graphs/jurkat.minus.bedGraph",
                "coverage_graphs/jurkat.minus.bigWig",
                "coverage_graphs/jurkat.plus.bedGraph",
                "coverage_graphs/jurkat.plus.bigWig",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/fastqc-status-check-heatmap.txt",
                "multiqc/multiqc_data/fastqc_adapter_content_plot.txt",
                "multiqc/multiqc_data/fastqc_overrepresented_sequences_plot.txt",
                "multiqc/multiqc_data/fastqc_per_base_n_content_plot.txt",
                "multiqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_counts_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt",
                "multiqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt",
                "multiqc/multiqc_data/homer-tag-directory-gc-content.txt",
                "multiqc/multiqc_data/homer-tag-info-dist.txt",
                "multiqc/multiqc_data/homer-tag-length-dist.txt",
                "multiqc/multiqc_data/homer_tagdir.txt",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_fastqc.txt",
                "multiqc/multiqc_data/multiqc_general_stats.txt",
                "multiqc/multiqc_data/multiqc_rseqc_infer_experiment.txt",
                "multiqc/multiqc_data/multiqc_rseqc_read_distribution.txt",
                "multiqc/multiqc_data/multiqc_samtools_flagstat.txt",
                "multiqc/multiqc_data/multiqc_samtools_idxstats.txt",
                "multiqc/multiqc_data/multiqc_samtools_stats.txt",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_data/preseq.txt",
                "multiqc/multiqc_data/preseq_complexity_plot_molecules.txt",
                "multiqc/multiqc_data/rseqc_infer_experiment_plot.txt",
                "multiqc/multiqc_data/rseqc_read_distribution_plot.txt",
                "multiqc/multiqc_data/rseqc_read_dups.txt",
                "multiqc/multiqc_data/rseqc_read_dups_plot.txt",
                "multiqc/multiqc_data/samtools-flagstat-dp_Percentage_of_total.txt",
                "multiqc/multiqc_data/samtools-flagstat-dp_Read_counts.txt",
                "multiqc/multiqc_data/samtools-idxstats-mapped-reads-plot_Normalised_Counts.txt",
                "multiqc/multiqc_data/samtools-idxstats-mapped-reads-plot_Observed_over_Expected_Counts.txt",
                "multiqc/multiqc_data/samtools-idxstats-mapped-reads-plot_Raw_Counts.txt",
                "multiqc/multiqc_data/samtools-stats-dp.txt",
                "multiqc/multiqc_data/samtools_alignment_plot.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_nascent_software_mqc_versions.yml",
                "preprocessing",
                "preprocessing/fastp",
                "preprocessing/fastp/cd4_REP1.trimmed.fastp.html",
                "preprocessing/fastp/cd4_REP1.trimmed.fastp.json",
                "preprocessing/fastp/cd4_REP1.trimmed.fastp.log",
                "preprocessing/fastp/cd4_REP2.trimmed.fastp.html",
                "preprocessing/fastp/cd4_REP2.trimmed.fastp.json",
                "preprocessing/fastp/cd4_REP2.trimmed.fastp.log",
                "preprocessing/fastp/jurkat.trimmed.fastp.html",
                "preprocessing/fastp/jurkat.trimmed.fastp.json",
                "preprocessing/fastp/jurkat.trimmed.fastp.log",
                "preprocessing/fastqc",
                "preprocessing/fastqc/cd4_REP1_fastqc.html",
                "preprocessing/fastqc/cd4_REP1_fastqc.zip",
                "preprocessing/fastqc/cd4_REP2_fastqc.html",
                "preprocessing/fastqc/cd4_REP2_fastqc.zip",
                "preprocessing/fastqc/jurkat_fastqc.html",
                "preprocessing/fastqc/jurkat_fastqc.zip",
                "quality_control",
                "quality_control/bbsplit",
                "quality_control/bbsplit/cd4_REP1.coverage.hist.txt",
                "quality_control/bbsplit/cd4_REP1.coverage.stats.txt",
                "quality_control/bbsplit/cd4_REP2.coverage.hist.txt",
                "quality_control/bbsplit/cd4_REP2.coverage.stats.txt",
                "quality_control/bbsplit/jurkat.coverage.hist.txt",
                "quality_control/bbsplit/jurkat.coverage.stats.txt",
                "quality_control/preseq",
                "quality_control/preseq/cd4_REP1.c_curve.txt",
                "quality_control/preseq/cd4_REP1.lc_extrap.txt",
                "quality_control/preseq/cd4_REP2.c_curve.txt",
                "quality_control/preseq/cd4_REP2.lc_extrap.txt",
                "quality_control/preseq/jurkat.c_curve.txt",
                "quality_control/preseq/jurkat.lc_extrap.txt",
                "quality_control/preseq/log",
                "quality_control/preseq/log/cd4_REP1.command.log",
                "quality_control/preseq/log/cd4_REP2.command.log",
                "quality_control/preseq/log/jurkat.command.log",
                "quality_control/rseqc",
                "quality_control/rseqc/infer_experiment",
                "quality_control/rseqc/infer_experiment/cd4_REP1.infer_experiment.txt",
                "quality_control/rseqc/infer_experiment/cd4_REP2.infer_experiment.txt",
                "quality_control/rseqc/infer_experiment/jurkat.infer_experiment.txt",
                "quality_control/rseqc/read_distribution",
                "quality_control/rseqc/read_distribution/cd4_REP1.read_distribution.txt",
                "quality_control/rseqc/read_distribution/cd4_REP2.read_distribution.txt",
                "quality_control/rseqc/read_distribution/jurkat.read_distribution.txt",
                "quality_control/rseqc/read_duplication",
                "quality_control/rseqc/read_duplication/pdf",
                "quality_control/rseqc/read_duplication/pdf/cd4_REP1.DupRate_plot.pdf",
                "quality_control/rseqc/read_duplication/pdf/cd4_REP2.DupRate_plot.pdf",
                "quality_control/rseqc/read_duplication/pdf/jurkat.DupRate_plot.pdf",
                "quality_control/rseqc/read_duplication/rscript",
                "quality_control/rseqc/read_duplication/rscript/cd4_REP1.DupRate_plot.r",
                "quality_control/rseqc/read_duplication/rscript/cd4_REP2.DupRate_plot.r",
                "quality_control/rseqc/read_duplication/rscript/jurkat.DupRate_plot.r",
                "quality_control/rseqc/read_duplication/xls",
                "quality_control/rseqc/read_duplication/xls/cd4_REP1.pos.DupRate.xls",
                "quality_control/rseqc/read_duplication/xls/cd4_REP1.seq.DupRate.xls",
                "quality_control/rseqc/read_duplication/xls/cd4_REP2.pos.DupRate.xls",
                "quality_control/rseqc/read_duplication/xls/cd4_REP2.seq.DupRate.xls",
                "quality_control/rseqc/read_duplication/xls/jurkat.pos.DupRate.xls",
                "quality_control/rseqc/read_duplication/xls/jurkat.seq.DupRate.xls",
                "quantification",
                "quantification/gene",
                "quantification/gene/cd4.featureCounts.txt",
                "quantification/gene/cd4.featureCounts.txt.summary",
                "quantification/gene/jurkat.featureCounts.txt",
                "quantification/gene/jurkat.featureCounts.txt.summary",
                "quantification/nascent",
                "quantification/nascent/cd4-group_cd4_intersect-transcripts.featureCounts.txt",
                "quantification/nascent/cd4-group_cd4_intersect-transcripts.featureCounts.txt.summary",
                "quantification/nascent/jurkat-group_cd4_intersect-transcripts.featureCounts.txt",
                "quantification/nascent/jurkat-group_cd4_intersect-transcripts.featureCounts.txt.summary",
                "samtools",
                "samtools/cd4.bam",
                "samtools/jurkat.bam",
                "transcript_identification",
                "transcript_identification/filtered",
                "transcript_identification/filtered/cd4_filtered.bed",
                "transcript_identification/filtered/jurkat_filtered.bed",
                "transcript_identification/homer",
                "transcript_identification/homer/cd4.bed",
                "transcript_identification/homer/cd4.bedGraph.gz",
                "transcript_identification/homer/cd4.peaks.txt",
                "transcript_identification/homer/cd4_tagdir",
                "transcript_identification/homer/cd4_tagdir/chr21.tags.tsv",
                "transcript_identification/homer/cd4_tagdir/genomeGCcontent.txt",
                "transcript_identification/homer/cd4_tagdir/tagAutocorrelation.txt",
                "transcript_identification/homer/cd4_tagdir/tagCountDistribution.txt",
                "transcript_identification/homer/cd4_tagdir/tagFreq.txt",
                "transcript_identification/homer/cd4_tagdir/tagFreqUniq.txt",
                "transcript_identification/homer/cd4_tagdir/tagGCcontent.txt",
                "transcript_identification/homer/cd4_tagdir/tagInfo.txt",
                "transcript_identification/homer/cd4_tagdir/tagLengthDistribution.txt",
                "transcript_identification/homer/jurkat.bed",
                "transcript_identification/homer/jurkat.bedGraph.gz",
                "transcript_identification/homer/jurkat.peaks.txt",
                "transcript_identification/homer/jurkat_tagdir",
                "transcript_identification/homer/jurkat_tagdir/chr21.tags.tsv",
                "transcript_identification/homer/jurkat_tagdir/genomeGCcontent.txt",
                "transcript_identification/homer/jurkat_tagdir/tagAutocorrelation.txt",
                "transcript_identification/homer/jurkat_tagdir/tagCountDistribution.txt",
                "transcript_identification/homer/jurkat_tagdir/tagFreq.txt",
                "transcript_identification/homer/jurkat_tagdir/tagFreqUniq.txt",
                "transcript_identification/homer/jurkat_tagdir/tagGCcontent.txt",
                "transcript_identification/homer/jurkat_tagdir/tagInfo.txt",
                "transcript_identification/homer/jurkat_tagdir/tagLengthDistribution.txt",
                "transcript_identification/homer/versions.yml",
                "transcript_identification/intersect",
                "transcript_identification/intersect/cd4_intersect.bed",
                "transcript_identification/intersect/jurkat_intersect.bed",
                "transcript_identification/merged",
                "transcript_identification/merged/cd4_merged.bed",
                "transcript_identification/merged/jurkat_merged.bed",
                "transcript_identification/pints",
                "transcript_identification/pints/cd4_chr21_1_unidirectional_peaks.bed",
                "transcript_identification/pints/jurkat_chr21_1_unidirectional_peaks.bed"
            ],
            [
                "cd4_REP1.sorted.bam.flagstat:md5,863e2d506d5cc4239af98a5f31bbc906",
                "cd4_REP1.sorted.bam.idxstats:md5,b1dd8bcbd23c53c21f0e11082d9315f2",
                "cd4_REP1.sorted.bam.stats:md5,70b7cd3040a9b7ca70e919afb4dc7fea",
                "cd4_REP2.sorted.bam.flagstat:md5,0fd86dbf8f799fad49ba471702979bdc",
                "cd4_REP2.sorted.bam.idxstats:md5,53204e4c6a9f68664087e4a8123be46a",
                "cd4_REP2.sorted.bam.stats:md5,c0e8d8d265fce71dcd43e43c93d29ef3",
                "jurkat.sorted.bam.flagstat:md5,fd5f02b0f02a407447b804b1d80f5421",
                "jurkat.sorted.bam.idxstats:md5,c61af0847c1ad76c06a8de2815975b32",
                "jurkat.sorted.bam.stats:md5,c9a6ec83d8fb1a14f53351c76cfcd24b",
                "cd4.bed:md5,0193e58943726af89bfd00e9da2536d8",
                "jurkat.bed:md5,cb6932229eea2e09f61d48d7dd397ae1",
                "cd4_REP1.dreg.bedGraph:md5,8948a8fa86d8f6d413b77983189ff56e",
                "cd4_REP1.minus.bedGraph:md5,2a1c34f9d9ef9ff1b9da7874b9e3aaad",
                "cd4_REP1.minus.bigWig:md5,5280319275c98dcce023779fa389884d",
                "cd4_REP1.plus.bedGraph:md5,1509ec3a921e3109c5914e1bcef8cf33",
                "cd4_REP1.plus.bigWig:md5,72ccab3173f2018a22a4b36841247ba2",
                "cd4_REP2.dreg.bedGraph:md5,29f865a5668fae4a52a41589ac2b3179",
                "cd4_REP2.minus.bedGraph:md5,c530bc34fa3ec7ac49e88ff65f9c2f92",
                "cd4_REP2.minus.bigWig:md5,5e748c794e037f441741f7f409c8c5ad",
                "cd4_REP2.plus.bedGraph:md5,a675141da2874ec08d91591e5ea8242b",
                "cd4_REP2.plus.bigWig:md5,08674d52e9eeb08807c33ed3e4b3d504",
                "jurkat.dreg.bedGraph:md5,c25a4fb095e9f7d6766a3ce33e08f7d8",
                "jurkat.minus.bedGraph:md5,8d5d9a41df6eb6c56b1bfd3f39dc1fc6",
                "jurkat.minus.bigWig:md5,d06c8015c996bf520fff17266fd01f84",
                "jurkat.plus.bedGraph:md5,6ed63e5983edaa74fb3965676efdb674",
                "jurkat.plus.bigWig:md5,7a02334f2c7300ffdb5a2253c0937390",
                "cd4_REP1.trimmed.fastp.json:md5,bd9a3344c1591d6be4d524451f9dca53",
                "cd4_REP2.trimmed.fastp.json:md5,bc6e3f9ff7835f220535cc393b8eb25f",
                "jurkat.trimmed.fastp.json:md5,65f7247a87479c7157c7357d575336e2",
                "cd4.bed:md5,b55e5290d78941f36c3d1ecfef8e0062",
                "cd4.peaks.txt:md5,bdcd2ec3a56a8a4a01ed19e17da003f2",
                "jurkat.bed:md5,383cfaf10535dbe5d7f47607e345f4cb",
                "jurkat.peaks.txt:md5,d4914194eca6f06aadfe7eed08ab1bb8",
                "versions.yml:md5,7c0dcd0a18b3c753def73b96cb825792",
                "cd4_merged.bed:md5,9ace0ca4f1544bb0949355a20de98e6b",
                "jurkat_merged.bed:md5,f1dde43c4ad9dec972ff9fa38cc6f2fe",
                "cd4_chr21_1_unidirectional_peaks.bed:md5,0193e58943726af89bfd00e9da2536d8",
                "jurkat_chr21_1_unidirectional_peaks.bed:md5,cb6932229eea2e09f61d48d7dd397ae1"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-28T21:53:21.62668"
    }
}